[FEATURE] Add a dedicated weekly CI workflow

oliverklee opened the pull request:
This workflow runs more jobs and variations than the regular CI workflow that is run in PRs and after a merge to `main`.

Fixes #1848

heiko-hardt commented:
Hi, as part of our monthly PR cleanup, I noticed this pull request has been open for more than 6 months. Is there still interest in moving this forward, or can we close it?
If it's still relevant, please let us know or update the status - otherwise we'd close it soon to keep things tidy.

Thanks and best regards!

oliverklee commented:
This is still needed.

oliverklee committed:
[FEATURE] Add a dedicated weekly CI workflow

This workflow runs more jobs and variations than the
regular CI workflow that is run in PRs and after a
merge to `main`.

Also drop the PHP linting jobs for "in-between" versions
from the regular CI job to achieve faster CI runs.

Fixes #1848
